react-app/src/redux/game/reducer.js

```diff
@@ -0,0 +1,57 @@
+import { calculateWinner } from '../../utils';
+
+const initialState = {
+  history: [
+    {
+      squares: Array(9).fill(null)
+    }
+  ],
+  stepNumber: 0,
+  xIsNext: true,
+  winner: null
+};
+
+const handleClick = (state, squarePosition) => {
+  const history = state.history.slice(0, state.stepNumber + 1);
+  const current = history[state.stepNumber];
+  const squares = [...current.squares];
+  const squareIsFilled = squares[squarePosition];
+  if (state.winner || squareIsFilled) return state;
+  squares[squarePosition] = state.xIsNext ? 'X' : 'O';
+  const winner = calculateWinner(squares);
+  return {
+    ...state,
+    history: history.concat([
+      {
+        squares
+      }
+    ]),
+    xIsNext: !state.xIsNext,
+    stepNumber: history.length,
+    winner
+  };
+};
+
+const jumpTo = (state, step) => {
```

Comment:
same

Reply:
Also, this functions are pretty big, how about moving them to gameActionUtils.js and export them from there